src/feature/dirauth/process_descs.c

```diff
@@ -304,13 +376,15 @@ dirserv_router_get_status(const routerinfo_t *router, const char **msg,
 /** Return true if there is no point in downloading the router described by
  * <b>rs</b> because this directory would reject it. */
 int
-dirserv_would_reject_router(const routerstatus_t *rs)
+dirserv_would_reject_router(const routerstatus_t *rs,
+                            const vote_routerstatus_t *vrs)
 {
   uint32_t res;
+  struct ed25519_public_key_t pk;
+  memcpy(&pk.pubkey, vrs->ed25519_id, ED25519_PUBKEY_LEN);
```

Comment:
Lets use `ed25519_pubkey_copy()`.

Reply:
We can't: ed25519_pubkey_copy() takes two ed25519_public_key_t pointers, but one of the arguments to this memcpy is a raw uint8_t array.